lib/headless.js

```diff
@@ -0,0 +1,31 @@
+const puppeteer = require('puppeteer');
+
+const headless = async (url, {timeout=8, debug=false}) => {
+  let download_link = '';
+
+  let opt = debug ? {headless: false, slowMo: 250 } : {}
+
+  const browser = await puppeteer.launch(opt);
+  const page = await browser.newPage();
+  console.log(`Open ${url}`)
+  await page.goto(url);
+
+  let saveLink = await page.evaluate(() => document.querySelector('.download-link a').href)
+  
+  await page.goto(saveLink);
+
+  return new Promise( (resolve) => {
+    setTimeout(async () => {
```

Comment:
I think this part still can be improved to not using timeout

Reply:
on progress ..
thanks for review